src/main/java/mc/spigot/weatherbylocation/ServerLocator.java

```diff
@@ -70,8 +84,43 @@ private static LocationData geolocateIpAddress(String ipAddress) throws IOExcept
         locationResult.city = jsonNode.get("city").asText();
         locationResult.latitude = jsonNode.get("lat").asDouble();
         locationResult.longitude = jsonNode.get("lon").asDouble();
+
+        // Save LocationData object to config.yml
+        saveDataToConfig(ipAddress,locationResult,config);
+
+        // Log result
+        logger.info(String.format("Server location identified as %s, %s, %s at coordinates (%.3f, %.3f)", locationResult.city, locationResult.region, locationResult.countryCode, locationResult.latitude, locationResult.longitude));
+        return locationResult;
+    }
+
+    private static LocationData locationDataFromConfig( FileConfiguration config) {
+        Logger logger = Logger.getLogger("WeatherByLocation");
+
+        // Create LocationData object and return it
+        LocationData locationResult = new LocationData();
+
+        locationResult.countryCode = config.getString("countryCode");
+        locationResult.region = config.getString("region");
+        locationResult.city = config.getString("city");
+        locationResult.latitude = config.getDouble("latitude");
+        locationResult.longitude = config.getDouble("longitude");
+
         // Log result
         logger.info(String.format("Server location identified as %s, %s, %s at coordinates (%.3f, %.3f)", locationResult.city, locationResult.region, locationResult.countryCode, locationResult.latitude, locationResult.longitude));
         return locationResult;
     }
+
+    private static void saveDataToConfig(String ipAddress, LocationData locationResult, FileConfiguration config) {
+
+        // Save new locationData object to config.yml file
+        config.set("ipAddress", ipAddress);
+        config.set("countryCode",  locationResult.countryCode);
+        config.set("region", locationResult.region);
+        config.set("city",  locationResult.city);
+        config.set("latitude", locationResult.latitude);
+        config.set("longitude",  locationResult.longitude);
+
```

Comment:
I needed to add this to get the plugin to save these values to the config file.

```suggestion
plugin.saveConfig();
```

Reply:
Aha i see, sorry about that mistake. I knew that Bukkit had an in-memory copy of the config file, but it seems that in order for it to be written to disk (the config.yml) we must explicitly save it as it does not automagically happen  [https://bukkit.fandom.com/wiki/Configuration_API_Reference]

Will not happen again

Reply:
No worries! I made the same mistake when first implementing the configuration file.